woocommerce/google-listings-and-ads
# Add free shipping order value condition for shipping phase 1

### Changes proposed in this Pull Request:



Part of pcTzPl-2qP-p2, this PR implements the last design of the new shipping phase 1 for setting the free shipping order value.

The PR follows [this design](https://www.figma.com/file/fqR0EHi63lWahRcVTKCcba?node-id=8367-25570&t=um2mPCfNxQjwBNpI-4) with slightly changes based on the discussion in pcTzPl-2qP-p2#comment-3723 and pcTzPl-2qP-p2#comment-3731.

Notable changes in this PR:

- Create a section for free shipping **Order value condition**, as previously it's part of the shipping rate section.
  - New folder `js/src/components/order-value-condition-section`
- Move `MinumumOrderCard` from `js/src/components/shipping-rate-section` to the new folder `js/src/components/order-value-condition-section`, since the card can be reused with slightly changes.
- Change the method [getCardProps()](https://github.com/woocommerce/google-listings-and-ads/blob/417385411a41fab54c8124875c5cbe8e5cf0d6fa/js/src/components/shipping-rate-section/flat-shipping-rates-input-cards.js#L16) as a shared hook in [js/src/components/adaptive-form/adaptive-form-context.js](https://github.com/woocommerce/google-listings-and-ads/pull/2686/files#diff-b1e1872c71d547b0a7fb7374740e842c18234bb4723ae7039571bce115479db4R75) as there are more than one place require to use this method.
  - I'm not sure if this is the best approach, feel free to suggest anything.


### Screenshots:



https://github.com/user-attachments/assets/5bcff49b-d705-4f65-b39d-2d2ecab6eb23



### Detailed test instructions:


1. Build the project using this PR
2. Start the onboarding flow, stop by step 2
3. Select **My shipping settings are simple. I can manually estimate flat shipping rates.** in **Shipping rates** section
4. Enter an estimated shipping rate, it should be > 0
5. See there is a new section **Order value condition** under **Shipping rates**
6. Set the free shipping order value condition for multiple target countries
7. Finish the onboarding flow
8. Make sure the free shipping order value conditions are synced to the Google Merchant Center
    - https://merchants.google.com/mc/shipping/services
9. Go to Dashboard, click **Edit** on the free listing
10. Redo from the step 3, and see it's working normally.

### Additional details:


### Changelog entry

>
